optas/builder.py

```diff
@@ -210,6 +212,104 @@ def get_model_states(self, name, time_deriv=0):
         return self._decision_variables[name]
 
 
+    def get_model_parameters(self, name, time_deriv=0):
+        """Get the vector of parameters for a given model.
+
+        Syntax
+        ------
+
+        parameters = builder.get_model_parameters(name, time_deriv=0)
+
+        Parameters
+        ----------
+
+        name (string)
+            Name of the model.
+
+        time_deriv (int)
+            The time-deriviative required (i.e. position is 0, velocity is 1, etc.)
+
+        Returns
+        -------
+
+        parameters (casadi.SX, with shape dim-by-T)
+            The vector of parameters where dim is the number of model parameters, and T is the number of time-steps in the trajectory.
+
+        """
+        model = self.get_model(name)
+        assert time_deriv in model.time_derivs, f"model '{name}', was not specified with time derivative to order {time_deriv}"
+        name = model.parameter_name(time_deriv)
+        return self._parameters[name]
+
+
+    def get_model_parameter(self, name, t, time_deriv=0):
+        """Get the model parameter at a given time.
+
+        Syntax
+        ------
+
+        parameter = builder.get_model_parameter(name, t, time_deriv=0)
+
+        Parameters
+        ----------
+
+        name (string)
+            Name of the model.
+
+        t (int)
+            Index of the desired parameter.
+
+        time_deriv (int)
+            The time-deriviative required (i.e. position is 0, velocity is 1, etc.)
+
+        Returns
+        -------
+
+        parameter (casadi.SX, with shape dim-by-1)
+            The parameter vector where dim is the model number of parameters
+
+        """
+        parameters = self.get_model_parameters(name, time_deriv)
+        return parameters[:, t]
+
+
+    def get_model_states_and_parameters(self, name, time_deriv=0):
+        """Get the vector of states and parameters for a given model.
+
+        Syntax
+        ------
+
+        parameters = builder.get_model_states_and_parameters(name, time_deriv=0)
+
+        Parameters
+        ----------
+
+        name (string)
+            Name of the model.
+
+        time_deriv (int)
+            The time-deriviative required (i.e. position is 0, velocity is 1, etc.)
+
+        Returns
+        -------
+
+        states and parameters (casadi.SX, with shape dim-by-T)
+            The vector of parameters where dim is the number of model states and parameters (for a robot it should correspond to the degrees of freedom), and T is the number of time-steps in the trajectory.
+
+        """
```

Comment:
I'm not sure I see the utility for this method. Also, I think it may break for different instantiations of the `OptimizationBuilder` (see below). I guess you want to be able to do something like `q = builder.get_joint_states()`, where the joint state `q` is a mix of decision variables and parameters?

Since this method only applies for `RobotModel`'s perhaps `robot` should appear in the method name since it will never be used for `TaskModel`'s. Then the `model` (from `get_model`) should be checked to ensure the user is referencing a robot model rather than a task model (you can use `assert` for this), something like

```python
assert isinstance(model, RobotModel), "this method only applies to robot models"
```

Reply:
The method is needed (I am using it) for not breaking backwards compatibility when you are using get_joint_states, which should still work as expected.
I changed the name to get_robot_states_and_parameters().

Reply:
Added the asset.